lib/internal/crypto/webcrypto.js

```diff
@@ -0,0 +1,727 @@
+    // the `NODE-` prefix
+    case 'NODE-DSA':
+      return lazyRequire('internal/crypto/dsa')
+        .dsaSignVerify(key, data, algorithm, signature);
+  }
+  throw lazyDOMException('Unrecognized named.', 'NotSupportedError');
+}
+
+async function sign(algorithm, key, data) {
+  return signVerify(algorithm, key, data);
+}
+
+async function verify(algorithm, key, signature, data) {
+  return signVerify(algorithm, key, data, signature);
+}
+
+async function cipherOrWrap(mode, algorithm, key, data, op) {
+  algorithm = normalizeAlgorithm(algorithm);
+  // We use a Node.js style error here instead of a DOMException because
+  // the WebCrypto spec is not specific what kind of error is to be thrown
+  // in this case. Both Firefox and Chrome throw simple TypeErrors here.
+  if (!isCryptoKey(key))
+    throw new ERR_INVALID_ARG_TYPE('key', 'CryptoKey', key);
+  // The key algorithm and cipher algorithm must match, and the
+  // key must have the proper usage.
+  if (key.algorithm.name !== algorithm.name ||
+      !ArrayPrototypeIncludes(key.usages, op)) {
+    throw lazyDOMException(
+      'The requested operation is not valid for the provided key',
+      'InvalidAccessError');
+  }
+
+  // For the WebCrypto API, the input data can be any ArrayBuffer,
+  // TypedArray, or DataView.
+  data = getArrayBufferOrView(data, 'data');
+
+  // While WebCrypto allows for larger input buffer sizes, we limit
+  // those to sizes that can fit within uint32_t because of limitations
+  // in the OpenSSL API.
+  validateMaxBufferLength(data, 'data');
+
+  switch (algorithm.name) {
+    case 'RSA-OAEP':
+      return lazyRequire('internal/crypto/rsa')
+        .rsaCipher(mode, key, data, algorithm);
+    case 'AES-CTR':
+      // Fall through
+    case 'AES-CBC':
+      // Fall through
+    case 'AES-GCM':
+      return lazyRequire('internal/crypto/aes')
+        .aesCipher(mode, key, data, algorithm);
+    case 'AES-KW':
+      if (op === 'wrapKey' || op === 'unwrapKey') {
+        return lazyRequire('internal/crypto/aes')
+          .aesCipher(mode, key, data, algorithm);
+      }
+  }
+  throw lazyDOMException('Unrecognized name.', 'NotSupportedError');
+}
+
+async function encrypt(algorithm, key, data) {
+  return cipherOrWrap(kWebCryptoCipherEncrypt, algorithm, key, data, 'encrypt');
+}
+
+async function decrypt(algorithm, key, data) {
+  return cipherOrWrap(kWebCryptoCipherDecrypt, algorithm, key, data, 'decrypt');
+}
+
+// The SubtleCrypto and Crypto classes are defined as part of the
+// WebCrypto API standard: https://www.w3.org/TR/WebCryptoAPI/
+
+class SubtleCrypto {}
+const subtle = new SubtleCrypto();
+
+class Crypto {}
+const crypto = new Crypto();
+
+ObjectDefineProperties(
+  Crypto.prototype, {
+    subtle: {
+      enumerable: true,
+      configurable: false,
+      value: subtle
+    },
+    getRandomValues: {
+      enumerable: true,
+      configurable: true,
+      writable: true,
+      value: getRandomValues
+    }
+  });
```

Comment:
```suggestion
ObjectDefineProperties(
  Crypto.prototype, {
      [SymbolToStringTag]: {
      enumerable: false,
      configurable: true,
      writable: false,
      value: 'Crypto'
    },
    subtle: {
      enumerable: true,
      configurable: true,
      get() { return subtle; },
    },
    getRandomValues: {
      enumerable: true,
      configurable: true,
      writable: true,
      value: getRandomValues
    }
  });
```

Both Chrome and Firefox define it like that.